src/dmd/dmodule.d

```diff
@@ -535,6 +501,77 @@ extern (C++) final class Module : Package
         return m;
     }
 
+    /**
+     * Converts a chain of identifiers to the filename of the module
+     *
+     * Params:
+     *  packages = the names of the "parent" packages
+     *  ident = the name of the child package or module
+     *
+     * Returns:
+     *  the filename of the child package or module
+     */
+    static const(char)* getFilename(Identifiers* packages, Identifier ident)
+    {
+        const(char)[] filename = ident.toString();
+
+        if (packages == null || packages.dim == 0)
+            return filename.ptr;
+
+        OutBuffer buf;
+        OutBuffer dotmods;
+        auto ms = global.params.modFileAliasStrings;
+        const msdim = ms ? ms.dim : 0;
+
+        void checkModFileAlias(const(char)[] p)
+        {
+            /* Check and replace the contents of buf[] with
+            * an alias string from global.params.modFileAliasStrings[]
+            */
+            dotmods.writestring(p);
+        Lmain:
+            for (size_t j = msdim; j--;)
+            {
+                const m = (*ms)[j];
+                const q = strchr(m, '=');
+                assert(q);
+                if (dotmods.offset == q - m && memcmp(dotmods.peekString(), m, q - m) == 0)
+                {
+                    buf.reset();
+                    auto qlen = strlen(q + 1);
```

Comment:
Having `qlen` be the length of string `q + 1` is wacky and makes the rest of the code hard to decipher because it is all one off from expected string handling code. Instead,
```
const r = q[1 .. strlen(q + 1)];
```
And then treat `r` like a normal string.